spec/index.html

```diff
@@ -421,6 +569,91 @@ <h2>Fragment identifiers</h2>
       <h3>Examples</h3>
       <p class="ednote">FIXME</p>
     </section>
+    <section id="faq" class="informative" data-format="markdown">
+      **NOTE** REMOVE THIS SECTION BEFORE PUBLICATION.
+
+      <h3>FAQ</h3>
+
+      #### Why YAML-LD does not preserve comments?
+      <p class="ednote">
+      [[JSON]] does not support comments.
+      While YAML-LD could define a specific predicate for that,
+      that is not enough because for example
+      in JSON the order of keywords is not preserved would
+      displace the comments.
+
+      ```yaml
+      # First comment
+      "@context": "http://schema.org"
+
+      # Second comment
+      givenName: John
+      ```
+      
+      transforming the above entry into a JSON-LD document
+      would result in:
+      
+      ```json
+      {
+        "@context": "http://schema.org",
+        "givenName": "John",
+        "http://www.w3.org/ns/json-ld/yaml-ld#comment": [
+          "First comment",
+          "Second comment"
+        ]
+      }
+      ```
+
+      
+      #### Why YAML-LD does not extend JSON-LD data structure ?
+      <p class="ednote">
+      [[JSON]] represent simple trees while [[YAML]] can support
+      rooted, directed graphs with references and cycles.
+
+      The above structures cannot be preserved when serializing
+      to JSON-LD and - with respect to cycles - they serialization
+      will fail.
+
+      Programming languages such as java an python already support
+      YAML representation graphs, but those implementation might behave
+      differently.
+      In the following example, `&value` references the value
+      of the keyword `value`.
+
+      ```yaml
+      value: &value 100
+      valve1:
+        temperature: &temp100C
+          value: *value
+          unit: degC
+      valve2:
+        temperature: *temp100C
+      ```
+
+      In python, processing this entry I get the following
+      structure, that preserves only the references to
+      mutable objects (e.g. the `temperature` dict)
+      but not the one to the `value` keyword (which is a scalar).
+
+      ```python
+      temperature = { "value": 100, "unit": "degC" }
+      document = {
+        "value": 100,
+        "valve1": { "temperature": temperature },
+        "valve2": { "temperature": temperature }
+      }
+      ```
+
+      Since all these implementations pre-date this 
+      specification, the more interoperable choices would be:
+      
+      * to forbid cycles in YAML-LD documents;
+      * to consider all references as static, e.g.
+        a shorthand way to repeat specific patterns.
+
+      </p>
+    </section>
+
```

Comment:
We should probably discuss how YAML dates and dateTime (equivalents) are treated. Presumably, as part of the extended profile, they can be expanded into their equivalent JSON-LD/XSD form, although considering the affect of context definitions for properties of which they are the value may complicate this somewhat.

Also, are YAML-LD processors required to parse all input regardless of the stated profile (expanded or basic), or only if the appropriate profile is set? I think the rule of profile parameters is that they don't change the semantics, so they either MUST process all forms, or error if the form is incorrect. (should be an issue).

Reply:
IIUC YAML 1.2 does not have a native date type anymore,  see https://github.com/yaml/yaml-spec/issues/268

Should be discussed after merging imho https://github.com/json-ld/yaml-ld/issues/35

Reply:
> IIUC YAML 1.2 does not have a native date type anymore, see [yaml/yaml-spec#268](https://github.com/yaml/yaml-spec/issues/268)

Just as well. But, it may come up again in the context of CBOR.